src/Products/Formulator/PatternChecker.py

```diff
@@ -57,25 +59,28 @@ def _replace_symbol_by_regex(self, match_object):
             min = length
             max = str(min)
 
-        return self.symbol_regex_dict[match_object.group(0)[0]] %(min, max)
+        return self.symbol_regex_dict[match_object.group(0)[0]] % (min, max)
 
     def make_regex_from_pattern(self, pattern):
         """Replaces all symbol occurences and creates a complete regex
         string.
         """
         regex = self._escape_special_characters(pattern)
         for symbol in [NUMBERSYMBOL, CHARSYMBOL, NUMCHARSYMBOL]:
-            regex = re.sub(symbol+'{1,}\*?', self._replace_symbol_by_regex, regex)
+            regex = re.sub(
+                symbol + r'{1,}\*?',
+                self._replace_symbol_by_regex,
+                regex)
         return '^ *' + regex + ' *$'
 
     def construct_value_from_match(self, result, pattern):
         """After we validated the string, we put it back together; this is
         good, since we can easily clean up the data this way.
         """
         pattern = self._escape_special_characters(pattern)
-        _symbols = '['+NUMBERSYMBOL + CHARSYMBOL + NUMCHARSYMBOL + ']'
-        re_obj = re.compile(_symbols+'{1,}\*?')
-        values=[]
+        _symbols = '[' + NUMBERSYMBOL + CHARSYMBOL + NUMCHARSYMBOL + ']'
+        re_obj = re.compile(_symbols + '{1,}\\*?')
```

Comment:
```suggestion
        re_obj = re.compile(_symbols + r'{1,}\*?')
```
In line 71 a raw string was used and I would rather prefer that as regexes are already hard to read on their own.